test/views/optOut/OptOutChooseTaxYearViewSpec.scala

```diff
@@ -83,4 +101,16 @@ class OptOutChooseTaxYearViewSpec extends TestSupport {
     }
 
   }
+
+  "Opt-out confirm page (with submissions)" should {
+
+    "show the warning block (for no submissions) user: non-agent" in new SetupNoSubmissions(false) {
+      pageDocument.getElementById("warning-inset") shouldBe null
+    }
+
+    "show the warning block (for no submissions) user: agent" in new SetupNoSubmissions(true) {
+      pageDocument.getElementById("warning-inset") shouldBe null
+    }
+  }
 }
```

Comment:
Think these tests can go, generally what I see is we can just ignore or not write tests if the component is not present. Also wording is off these tests I'm thinking would say  "not show"/"not present".

Also avoid using nulls in Scala there is almost never a need to use them. 
If we had to add a test for a element not being present we can likely do either:

1. Wrap the element in an Option and then check it is None, tbh it's kinda samey as throwing null.
2. Use css selectors think .select("#selector-goes-here") or some other methods like .getElements... return a Seq/List and we can check for an empty List()/Seq().

But yeah would go with deleting the test.

Reply:
happy to delete them, are we sure we want to go that route for dynamic text though? Just thinking if the logic gets tampered with in the future and starts showing all the time for whatever reason, we may not know as a test wouldn't flag - just a thought